src/sources/connect/query_graph.rs

```diff
@@ -0,0 +1,812 @@
+use apollo_compiler::{
+    ast::{FieldDefinition, Name},
+    schema::Component,
+};
+use indexmap::{map::Entry, IndexMap, IndexSet};
+use itertools::Itertools;
+use petgraph::prelude::NodeIndex;
+
+use crate::{
+    error::FederationError,
+    schema::{position::TypeDefinitionPosition, ValidFederationSchema},
+    source_aware::federated_query_graph::builder::IntraSourceQueryGraphBuilderApi,
+    sources::{
+        connect::ConnectFederatedScalarQueryGraphNode, FederatedLookupTailData,
+        SourceFederatedConcreteFieldQueryGraphEdge, SourceFederatedConcreteQueryGraphNode,
+        SourceFederatedEnumQueryGraphNode, SourceFederatedQueryGraphBuilderApi,
+        SourceFederatedScalarQueryGraphNode,
+    },
+    ValidFederationSubgraph,
+};
+
+use super::{
+    selection_parser::{NamedSelection, PathSelection, Property, SubSelection},
+    spec::schema::{ConnectDirectiveArguments, CONNECT_DIRECTIVE_NAME_IN_SPEC},
+    ConnectFederatedConcreteFieldQueryGraphEdge, ConnectFederatedConcreteQueryGraphNode,
+    ConnectFederatedEnumQueryGraphNode, ConnectFederatedQueryGraphBuilder, Selection,
+};
+
+impl SourceFederatedQueryGraphBuilderApi for ConnectFederatedQueryGraphBuilder {
+    fn process_subgraph_schema(
+        &self,
+        subgraph: ValidFederationSubgraph,
+        builder: &mut impl IntraSourceQueryGraphBuilderApi,
+    ) -> Result<Vec<FederatedLookupTailData>, FederationError> {
+        // Grab all connects
+        // TODO: Handle renames
+        let connect_references = subgraph
+            .schema
+            .referencers()
+            .get_directive(&CONNECT_DIRECTIVE_NAME_IN_SPEC)?;
+
+        // Extract the connects from the schema definition and map them to their `Connect` equivalent
+        // Note: We need to keep the object_field as well as the arguments since it contains important
+        // type information of the original schema.
+        // TODO: Validate that we don't have connect directives on anything but object fields.
+        let connects: Vec<_> = connect_references
+            .object_fields
+            .iter()
+            .map(|object_field| {
+                // Note: We need to keep the `object_field` around since it contains metadata needed for graph
+                // creation.
+                object_field
+                    .get(subgraph.schema.schema())
+                    .map(move |f| (object_field, f))
+            })
+            .map_ok(|(object_field, field)| {
+                field
+                    .directives
+                    .iter()
+                    .map(move |d| (object_field, field, d))
+            })
+            .flatten_ok()
+            .map_ok(|(object_field, field, directive)| {
+                ConnectDirectiveArguments::try_from(directive)
+                    .map(move |args| (object_field, field, args))
+            })
+            .flatten_ok()
+            .try_collect()?;
```

Comment:
let's move this into a separate function — i think i'd like to process the connect and source directives into a normalized representation, _and then_ validate and process into a query graph separately

Reply:
Will be done by validation work separate to this PR